skopt/gp_opt.py

```diff
@@ -161,13 +163,15 @@ def gp_minimize(func, dimensions, base_estimator=None, acq="EI", xi=0.01,
         For more details related to the OptimizeResult object, refer
         http://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.OptimizeResult.html
     """
-    # Save call args 
+    # Save call args
     specs = {"args": copy.copy(inspect.currentframe().f_locals),
              "function": inspect.currentframe().f_code.co_name}
 
     # Check params
     rng = check_random_state(random_state)
     space = Space(dimensions)
+    is_discrete_space = all(
```

Comment:
Not sure I agree with this definition of discrete space. We should be clearer. `Integer` are discrete. Same if only one of the dimensions (not all) is `Categorical` or `Integer`.